audit.py

```diff
@@ -47,8 +47,160 @@ def unknown(what: str) -> None:
 output_option = typer.Option(None, "--output", "-o")
 
 
-g = Github(config["github.com"]["token"])
-app = typer.Typer()
+class ReportBase:
+
+    def __init__(self, started, output):
+        self.started = started
+        self.output = output
+
+    def begin_report(self, title: str):
+        raise NotImplementedError(f"{self.__class__.__name__}.begin_report()")
+
+    def end_report(self):
+        raise NotImplementedError(f"{self.__class__.__name__}.end_report()")
+
+    def empty_line(self):
+        raise NotImplementedError(f"{self.__class__.__name__}.empty_line()")
+
+    def begin_table(self):
+        raise NotImplementedError(f"{self.__class__.__name__}.begin_table()")
+
+    def table_row(self, *args, **kwargs):
+        raise NotImplementedError(f"{self.__class__.__name__}.table_row()")
+
+    def end_table(self):
+        raise NotImplementedError(f"{self.__class__.__name__}.end_table()")
+
+
+class TextReportBase(ReportBase):
+
+    table = None
```

Comment:
This could use a comment explaining why table is not an instance attribute.

Reply:
Open to discussion best avoided - fixed